app/ui/legacy/src/main/java/com/fsck/k9/activity/setup/AuthViewModel.kt

```diff
@@ -88,54 +83,34 @@ class AuthViewModel(
         val account = checkNotNull(account)
 
         viewModelScope.launch {
-            val config = findOAuthConfiguration(account)
-            if (config == null) {
-                _uiState.update { AuthFlowState.NotSupported }
-                return@launch
-            }
-
             try {
-                startLogin(account, config)
+                startLogin(account)
             } catch (e: ActivityNotFoundException) {
                 _uiState.update { AuthFlowState.BrowserNotFound }
             }
         }
     }
 
-    private suspend fun startLogin(account: Account, config: OAuthConfiguration) {
+    private suspend fun startLogin(account: Account) {
         val authRequestIntent = withContext(Dispatchers.IO) {
-            createAuthorizationRequestIntent(account.email, config)
+            getOAuthRequestIntent.execute(account.incomingServerSettings.host!!, account.email)
+                .let { result ->
+                    when (result) {
+                        is DomainContract.UseCase.GetOAuthRequestIntent.GetOAuthRequestIntentResult.NotSupported -> {
```

Comment:
That looks excessive. Maybe add an import for `DomainContract.UseCase.GetOAuthRequestIntent.GetOAuthRequestIntentResult`?